client-ts/Microsoft.AspNetCore.SignalR.Test.Server/wwwroot/js/hubConnectionTests.js

```diff
@@ -62,7 +74,12 @@ describe('hubConnection', function () {
 
             it('rethrows an exception from the server when invoking', function (done) {
                 var errorMessage = "An error occurred.";
-                var hubConnection = new signalR.HubConnection(new signalR.HttpConnection(TESTHUBENDPOINT_URL, { transport: transportType }), { protocol: protocol});
+                var options = {
+                    transport: transportType,
+                    protocol: protocol,
+                    logging: signalR.LogLevel.Trace
+                };
+                var hubConnection = new signalR.HubConnection(new signalR.HttpConnection(TESTHUBENDPOINT_URL, options), options);
```

Comment:
I hate how hard (and weird (look at the `options), options);`) it is to create hubConnection in ts/js. 
Should we change the API so that IConnection is part of options and if not passed we would create an HttpConnection for the user?

Reply:
We need the builder api on the js client too...

Reply:
... or blur the separation a little bit as per https://github.com/aspnet/SignalR/issues/825